### PR TITLE
[WIP] Add support to FFI

### PR DESCRIPTION
**Pull request by dougsland**
Just an initial code for FFI:

- Add C sources to eat memory (ASIL: 20%) (QM: 90%)
- ASIL container should create it's own file from template

### CODE REVIEW

**Comment by dougsland** [comment]
For the disks spaces we can even use fallocate tool, example:

```
fallocate -l 300G $filename
```

**Review by Yarboa — COMMENTED** [review]
I would advise adding tests inside tmt tests and not inside the scripts
Consider copying the files while preparing the test,
under prepare of test.fmf 
You can also make the files in that stage

File: tests/e2e/lib/ContainerFile.template [hunk 1]
```diff
@@ -110,6 +110,10 @@ RUN sed -e '/^#ManagerPort=/s/^#//g' \
 	-e '/#LogTarget/s/^#//' \
 	-e "s/^#AllowedNodeNames=/AllowedNodeNames=control,node1/" \
 	-i /etc/bluechi/bluechi.conf
+
+RUN mkdir -p /root/tests/FFI
+COPY tools/FFI/memory/ASIL/20_percent_memory_eat.c /root/tests/FFI
+RUN cd /root/tests/FFI && gcc -o 20_percent_memory_eat 20_percent_memory_eat.c
```
Comment (Yarboa) [inline tests/e2e/lib/ContainerFile.template:116]: This part   should run under test prepare stage

File: tests/e2e/lib/container [hunk 2]
```diff
@@ -112,6 +112,10 @@ create_qm_node() {
                 ' /etc/bluechi/agent.conf' >> ContainerFile.node"${nodeID}"
         echo "RUN systemctl enable bluechi-agent &> /dev/null" >> ContainerFile.node"${nodeID}"
 
+        echo "RUN mkdir -p /usr/lib/qm/rootfs/root/tests/FFI" >> ContainerFile.node"${nodeID}"
```
Comment (Yarboa) [inline tests/e2e/lib/container:115]: dame as above

**Comment by dougsland** [comment]
based on the talk with @Yarboa , he will adapt this PR to something more aligned to QE. Thanks @Yarboa

**Pull request closed without merging**